CMakeLists.txt

```diff
@@ -224,6 +224,14 @@ if(MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX)
   message(STATUS "MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX selected, removing AVX optimisations")
 endif()
 
+# Some assemblers know about AVX but not AVX512 instructions, e.g. gcc 4.8.2.
+# This flag can be set to handle such cases.
+option(MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX512 "Exclude AVX512 code from the build" OFF)
```

Comment:
Having `MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX` set should probably automatically set `MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX512` as well.

Reply:
I thought of that, but we anyway need this section, right? For the cases that disable only AVX512. So NO AVX implies NO AVX512 but not the other way around, so do I just add the AVX512 definition to the section above?
In the AVX512 AESNI perl file, do I just check for AVX512?

Reply:
> I thought of that, but we anyway need this section, right? 

yes

> do I just add the AVX512 definition to the section above?

yes, you can just set `MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX512` variable to `ON`.

> In the AVX512 AESNI perl file, do I just check for AVX512?

I'd leave both checks, there is no cost to it.